lib/docx/document.rb

```diff
@@ -20,9 +20,11 @@ module Docx
   class Document
     attr_reader :xml, :doc, :zip, :styles
 
-    def initialize(path, &block)
+    def initialize(path_or_io, options = {})
       @replace = {}
-      @zip = Zip::File.open(path)
+
+      @zip = options[:use_buffer] ? Zip::File.open_buffer(path_or_io) : Zip::File.open(path_or_io)
```

Comment:
@visoft your idea looks good. but users can pass `use_buffer: true` with a file path. it should be prevented if possible. 
This means that `use_buffer` option can be removed if we have a way to detect whether `path_or_io` is a `IO` or a `String`.